lib/grp.gd

```diff
@@ -729,6 +729,43 @@ InstallTrueMethod( IsGroup and IsNonTrivial, IsAlmostSimpleGroup );
 # nonabelian simple groups are almost simple
 InstallTrueMethod( IsAlmostSimpleGroup, IsNonabelianSimpleGroup );
 
+#############################################################################
+##
+#P  IsQuasisimpleGroup( <G> )
+##
+##  <#GAPDoc Label="IsQuasisimpleGroup">
+##  <ManSection>
+##  <Prop Name="IsQuasisimpleGroup" Arg='G'/>
+##
+##  <Description>
+##  A group <A>G</A> is <E>quasisimple</E> if <A>G</A> is perfect
+##  (see <Ref Prop="IsPerfectGroup"/>)
+##  and if <A>G</A><M>/Z(</M><A>G</A><M>)</M> is simple
+##  (see <Ref Prop="IsSimpleGroup"/>), where <M>Z(</M><A>G</A><M>)</M>
+##  is the centre of <A>G</A> (see <Ref Attr="Centre"/>).
+##  <P/>
+##  <Example><![CDATA[
+##  gap> IsQuasisimpleGroup( AlternatingGroup( 5 ) );
+##  true
+##  gap> IsQuasisimpleGroup( SymmetricGroup( 5 ) );
+##  false
+##  gap> IsQuasisimpleGroup( SL( 2, 5 ) );
+##  true
+##  ]]></Example>
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+DeclareProperty( "IsQuasisimpleGroup", IsGroup );
+InstallTrueMethod( IsGroup and IsNonTrivial, IsQuasisimpleGroup );
+
+# Nonabelian simple groups are quasisimple, quasisimple groups are perfect.
+InstallTrueMethod( IsQuasisimpleGroup, IsNonabelianSimpleGroup );
+InstallTrueMethod( IsPerfectGroup, IsQuasisimpleGroup );
+
+# We can expect that people will try the name with capital s.
+DeclareSynonymAttr( "IsQuasiSimpleGroup", IsQuasisimpleGroup );
```

Comment:
Hmmm, indeed...  In particular since it is `IsAlmostSimpleGroup` and `IsNonabelianSimpleGroup`. What's the logic here? Is it because "quasisimple" is one word but "almost simple" is two?

So I guess for `IsNonabelianSimpleGroup ` someone had `nonabelian` in mind, while for `NonAbelianTensorSquare` (in packages) someone else uses `non-abelian`... *sigh*

We don't seem to provide such "up to case synonyms" for other things, though. Should we really start this here? If we do, I'd add at least a comment explaining why we do this here but not in other cases, e.g. the above reasoning (assuming it is what you had in mind).

Reply:
I thought that "camel case" means that the first letter of each word is capitalized. In the current situation, both quasisimple (one word) and quasi-simple (two words?) are common. The easiest way to deal with this situation is to support both names. In my eyes, this is analogous to `Centre` vs. `Center` and `Centralizer` vs. `Centraliser`.